app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -233,7 +236,7 @@ private HPlusSupport setCurrentDate(TransactionBuilder transaction) {
                 HPlusConstants.COMMAND_SET_PREF_DATE,
                 (byte) ((year / 256) & 0xff),
                 (byte) (year % 256),
-                (byte) (month),
+                (byte) (month + 1),
```

Comment:
I actually thought that you might need the offset, but I thought you'd know if the device shows the correct date :-)